examples/openai-function-call-example/go.mod

```diff
@@ -4,7 +4,7 @@ go 1.21
 
 toolchain go1.21.4
 
-require github.com/tmc/langchaingo v0.1.7
+require github.com/tmc/langchaingo v0.1.8-0.20240324221257-f3c3fcc40b49
```

Comment:
nit: the tag doesn't exist yet so this won't work for others

you're probably testing this locally with a go.work file? so this line can be omitted for now and later all examples can be updated with the script when 0.1.8 is tagged?

Reply:
This is pointing at this branch temporarily so this can build ahead of these changes merging.